substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/range/PrimaryRange.java

```diff
@@ -0,0 +1,147 @@
+     * parameter.
+     */
+    private DebugInfoProvider.DebugLocalInfo[] localInfos;
+
+    protected PrimaryRange(MethodEntry methodEntry, int lo, int hi, int line) {
+        super(methodEntry, lo, hi, line, -1);
+        this.firstCallee = null;
+        this.lastCallee = null;
+    }
+
+    @Override
+    public int getFileIndex() {
+        ClassEntry owner = methodEntry.ownerType();
+        return owner.localFilesIdx(getFileEntry());
+    }
+
+    @Override
+    public boolean isPrimary() {
+        return true;
+    }
+
+    @Override
+    protected void addCallee(SubRange callee) {
+        assert this.lo <= callee.lo;
+        assert this.hi >= callee.hi;
+        assert callee.caller == this;
+        assert callee.siblingCallee == null;
+        if (this.firstCallee == null) {
+            assert this.lastCallee == null;
+            this.firstCallee = this.lastCallee = callee;
+        } else {
+            this.lastCallee.siblingCallee = callee;
+            this.lastCallee = callee;
+        }
+    }
+
+    @Override
+    public SubRange getFirstCallee() {
+        return firstCallee;
+    }
+
+    @Override
+    public boolean isLeaf() {
+        return firstCallee == null;
+    }
+
+    @Override
+    public boolean includesInlineRanges() {
+        SubRange child = firstCallee;
+        while (child != null && child.isLeaf()) {
+            child = child.siblingCallee;
+        }
+        return child != null;
+    }
+
+    public int getLocalValueCount() {
+        assert !this.isPrimary() : "primary range does not have local values";
+        return localValueInfos.length;
+    }
+
+    public DebugInfoProvider.DebugLocalValueInfo getLocalValue(int i) {
+        assert !this.isPrimary() : "primary range does not have local values";
+        assert i >= 0 && i < localValueInfos.length : "bad index";
+        return localValueInfos[i];
+    }
+
+    public DebugInfoProvider.DebugLocalInfo getLocal(int i) {
+        assert !this.isPrimary() : "primary range does not have local vars";
+        assert i >= 0 && i < localInfos.length : "bad index";
+        return localInfos[i];
+    }
+
+    public void setLocalValueInfo(DebugInfoProvider.DebugLocalValueInfo[] localValueInfos) {
+        int len = localValueInfos.length;
+        this.localValueInfos = localValueInfos;
+        this.localInfos = (len > 0 ? new DebugInfoProvider.DebugLocalInfo[len] : EMPTY_LOCAL_INFOS);
+        // set up mapping from local values to local variables
+        for (int i = 0; i < len; i++) {
+            localInfos[i] = methodEntry.recordLocal(localValueInfos[i]);
+        }
+    }
+
+    public DebugInfoProvider.DebugLocalValueInfo lookupValue(DebugInfoProvider.DebugLocalInfo local) {
+        int localValueCount = getLocalValueCount();
+        for (int i = 0; i < localValueCount; i++) {
+            if (getLocal(i) == local) {
+                return getLocalValue(i);
+            }
+        }
+        return null;
+    }
```

Comment:
Since primary ranges do not have local values and vars these methods seem wrong

Reply:
Yes, they are gone.